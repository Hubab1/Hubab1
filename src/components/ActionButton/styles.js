import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
