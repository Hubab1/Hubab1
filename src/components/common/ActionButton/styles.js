import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
