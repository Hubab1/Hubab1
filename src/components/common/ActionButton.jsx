import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}