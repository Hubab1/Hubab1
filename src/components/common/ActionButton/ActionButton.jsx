import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}