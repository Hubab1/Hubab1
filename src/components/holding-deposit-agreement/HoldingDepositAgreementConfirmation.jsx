import React, {Fragment, useEffect, useState} from 'react';
import styled from '@emotion/styled';

import { H1, SpacedH3, Spacer } from 'assets/styles';
import ActionButton from 'components/common/ActionButton/ActionButton';

import contract from 'assets/images/contract.svg';
import {css} from 'emotion';
import API from 'app/api';
import { DOCUMENT_TYPE_HOLDING_DEPOSIT } from 'app/constants';
import GenericFormMessage from 'components/common/GenericFormMessage';

export const Img = styled.img`
    padding-top: 14px;
    height: 157px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export const HoldingDepositAgreementConfirmation = ({profile, configuration, handleContinue, viewDocument}) => {
    const [url, setUrl] = useState(null);
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState(false);
    const [retried, setRetried] = useState(false);

    const fetchLeaseDocumentUrl = async ()=>{
        setLoading(true);
        const response = await API.leaseDocumentUrl(DOCUMENT_TYPE_HOLDING_DEPOSIT);
        setUrl(response ? response.url : undefined);
        setError(response.url ? undefined : 'Holding Deposit Agreement is still processing. Please try again later.');
        setLoading(false);
    };

    useEffect(() => {
        fetchLeaseDocumentUrl();
    }, []);

    const onClick = () => {
        if (!url) {
            setRetried(true);
            fetchLeaseDocumentUrl();
        }
    };

    const getButtonText = () => {
        if (loading) {
            return 'Loading...';
        }
        if (url) {
            return 'Review Holding Deposit Agreement';
        }
        return 'Retrieve Holding Deposit Agreement...';
    };

    if (!profile || ! configuration) return null;

    const {
        unit,
    } = profile;
    const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
    const unitNumber = (!!unit && !!unit.unit_number) ? ` Unit ${unit.unit_number}` : '';

    return (
        <Fragment>
            <H1>Thanks for Signing!</H1>
            <SpacedH3>We&apos;ll send you an email with a copy of the signed agreement.</SpacedH3>
            <Img src={contract}/>
            <div className={applicationUnit}>{buildingName}{unitNumber}</div>
            <Spacer height={40}/>
            {retried && !!error && <GenericFormMessage type="error" messages={error}/>}
            <ActionButton onClick={handleContinue} marginTop={30} marginBottom={20}>Continue</ActionButton>
            <ActionButton disabled={loading} onClick={url ? undefined : onClick} href={url} variant="outlined" marginBottom={20}>
                {getButtonText()}
            </ActionButton>
        </Fragment>
    );
};

export default HoldingDepositAgreementConfirmation;
