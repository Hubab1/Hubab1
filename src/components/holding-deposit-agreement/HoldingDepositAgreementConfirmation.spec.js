import React from 'react';
import { mount } from 'enzyme';
import { HoldingDepositAgreementConfirmation } from "./HoldingDepositAgreementConfirmation";
import { act } from "react-dom/test-utils";
import ActionButton from 'components/common/ActionButton/ActionButton';
import API from 'app/api';


let defaultProps;

beforeEach(() => {
    defaultProps = {
        handleContinue: jest.fn(),
        viewDocument: jest.fn(),
        configuration: {
            community: {
                buildingName: 'Monterey Pines Apartments',

            }
        },
        profile: {
            unit: {
                unit_number: 'Unit 14F',
            }
        }
    };
});

it('Matches Snapshot', async () => {
    let wrapper = mount( <HoldingDepositAgreementConfirmation {...defaultProps}/> );
    expect(wrapper.debug()).toMatchSnapshot();
});

describe('Button text', function () {
    it('lets you refetch holding deposit url', async () => {
        API.leaseDocumentUrl = jest.fn().mockReturnValue({error: 'bad'});
        let wrapper = mount(<HoldingDepositAgreementConfirmation {...defaultProps} />);
        await act(async () => {
            await Promise.resolve(wrapper);
            wrapper.update();
        });
        expect(wrapper.find(ActionButton).last().prop('href')).toEqual(undefined);
        expect(wrapper.find(ActionButton).last().text()).toEqual('Retrieve Holding Deposit Agreement...');
    });

    it('fetches and links pdf url', async () => {
        API.leaseDocumentUrl = jest.fn().mockReturnValue({url: 'testpdfurl.pdf'});
        let wrapper = mount(<HoldingDepositAgreementConfirmation {...defaultProps} />);
        await act(async () => {
            await Promise.resolve(wrapper);
            wrapper.update();
        });
        expect(wrapper.find(ActionButton).last().prop('href')).toEqual('testpdfurl.pdf');
        expect(wrapper.find(ActionButton).last().text()).toEqual('Review Holding Deposit Agreement');
    });

    it('Displays loading state', async () => {
        let wrapper = mount(<HoldingDepositAgreementConfirmation {...defaultProps} />);
        expect(wrapper.find(ActionButton).last().prop('href')).toEqual(null);
        expect(wrapper.find(ActionButton).last().text()).toEqual('Loading...');
    });
});