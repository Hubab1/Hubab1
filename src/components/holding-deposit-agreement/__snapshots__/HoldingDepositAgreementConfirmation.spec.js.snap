// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
"<HoldingDepositAgreementConfirmation handleContinue={[Function: mockConstructor]} viewDocument={[Function: mockConstructor]} configuration={{...}} profile={{...}}>
  <Styled(h1)>
    <h1 className=\\"css-r7wuwd\\">
      Thanks for Signing!
    </h1>
  </Styled(h1)>
  <Styled(h3)>
    <h3 className=\\"css-1caxa3\\">
      We&#39;ll send you an email with a copy of the signed agreement.
    </h3>
  </Styled(h3)>
  <Styled(img) src=\\"contract.svg\\">
    <img src=\\"contract.svg\\" className=\\"css-1sepv4a\\" />
  </Styled(img)>
  <div className=\\"css-lygmaw\\">
     Unit Unit 14F
  </div>
  <Styled(div) height={40}>
    <div height={40} className=\\"css-vbdlzz\\" />
  </Styled(div)>
  <ActionButton onClick={[Function: mockConstructor]} marginTop={30} marginBottom={20} variant=\\"contained\\" color=\\"primary\\" successGreen={false}>
    <Styled(div) target=\\"_blank\\" href={[undefined]} as=\\"div\\" marginTop={30} marginBottom={20}>
      <div target=\\"_blank\\" href={[undefined]} className=\\"css-g5qjtx\\">
        <WithStyles(ForwardRef(Button)) onClick={[Function: mockConstructor]} classes={{...}} variant=\\"contained\\" color=\\"primary\\" type=\\"submit\\" disabled={[undefined]} fullWidth={true}>
          <ForwardRef(Button) classes={{...}} onClick={[Function: mockConstructor]} variant=\\"contained\\" color=\\"primary\\" type=\\"submit\\" disabled={[undefined]} fullWidth={true}>
            <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiButton-root css-11ldqzb MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"Mui-focusVisible\\" type=\\"submit\\" onClick={[Function: mockConstructor]}>
              <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiButton-root css-11ldqzb MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"Mui-focusVisible\\" type=\\"submit\\" onClick={[Function: mockConstructor]}>
                <button className=\\"MuiButtonBase-root MuiButton-root css-11ldqzb MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth\\" onBlur={[Function (anonymous)]} onClick={[Function: mockConstructor]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"submit\\" disabled={false}>
                  <span className=\\"MuiButton-label css-1405lf9\\">
                    Continue
                  </span>
                  <WithStyles(memo) center={false}>
                    <ForwardRef(TouchRipple) classes={{...}} center={false}>
                      <span className=\\"MuiTouchRipple-root\\">
                        <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                      </span>
                    </ForwardRef(TouchRipple)>
                  </WithStyles(memo)>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(Button)>
        </WithStyles(ForwardRef(Button))>
      </div>
    </Styled(div)>
  </ActionButton>
  <ActionButton disabled={true} onClick={[Function: onClick]} href={{...}} variant=\\"outlined\\" marginBottom={20} color=\\"primary\\" successGreen={false}>
    <Styled(div) target=\\"_blank\\" href={{...}} as=\\"div\\" marginTop={[undefined]} marginBottom={20}>
      <div target=\\"_blank\\" href={{...}} className=\\"css-4vllap\\">
        <WithStyles(ForwardRef(Button)) onClick={[Function: onClick]} classes={{...}} variant=\\"outlined\\" color=\\"primary\\" type=\\"submit\\" disabled={true} fullWidth={true}>
          <ForwardRef(Button) classes={{...}} onClick={[Function: onClick]} variant=\\"outlined\\" color=\\"primary\\" type=\\"submit\\" disabled={true} fullWidth={true}>
            <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiButton-root css-11ldqzb MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled css-e1321a MuiButton-fullWidth\\" component=\\"button\\" disabled={true} focusRipple={true} focusVisibleClassName=\\"Mui-focusVisible\\" type=\\"submit\\" onClick={[Function: onClick]}>
              <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiButton-root css-11ldqzb MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled css-e1321a MuiButton-fullWidth\\" component=\\"button\\" disabled={true} focusRipple={true} focusVisibleClassName=\\"Mui-focusVisible\\" type=\\"submit\\" onClick={[Function: onClick]}>
                <button className=\\"MuiButtonBase-root MuiButton-root css-11ldqzb MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled css-e1321a MuiButton-fullWidth Mui-disabled\\" onBlur={[Function (anonymous)]} onClick={[Function: onClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"submit\\" disabled={true}>
                  <span className=\\"MuiButton-label css-1405lf9\\">
                    Review Holding Deposit Agreement
                  </span>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(Button)>
        </WithStyles(ForwardRef(Button))>
      </div>
    </Styled(div)>
  </ActionButton>
</HoldingDepositAgreementConfirmation>"
`;
